src/App.js

```diff
@@ -0,0 +1,109 @@
+import { Component } from 'react';
+import AppHeader from './AppHeader';
+import ItemsList from './ItemsList';
+import AppFooter from './AppFooter';
+
+class App extends Component {
+    constructor() {
+        super()
+        this.state = {
+            inputfield: '',
+            inputs: [],
+            saveButtonClassActive: false
+        };
+    }
+
+    componentDidMount() {
+        const storageItems = localStorage.getItem('items');
+        const parsedItems = JSON.parse(storageItems);
```

Comment:
`JSON.parse` may throw an error if there would be an invalid JSON in localStorage. I know, that you may think that how there can get some invalid JSON, but in reality, you may use some libraries which store data in localStorage, or the user may have some extension that writes data to the localStorage.

You should always take care of the case where there some invalid (or no) data in localStorage.

Please wrap this in try-catch and ensure that invalid data would be handled properly.

Reply:
Ok, I have wrapped it in try-catch. But what should happen inside of catch? Just alert the error?

Reply:
that's a good question, common approach would be to put an error into the component state if occur and then show some nice UI message saying that something went wrong, and unfortunately we cannot restore saved todoItems